Client/src/Pages/PageSpeed/Details/Charts/AreaChart.jsx

```diff
@@ -0,0 +1,313 @@
+ * @returns {JSX.Element|null} The tooltip element or null if not active.
+ */
+
+const CustomToolTip = ({ active, payload, label, config }) => {
+  const theme = useTheme();
+  console.log(payload);
+  if (active && payload && payload.length) {
+    return (
+      <Box
+        sx={{
+          backgroundColor: theme.palette.background.main,
+          border: 1,
+          borderColor: theme.palette.border.dark,
+          borderRadius: theme.shape.borderRadius,
+          py: theme.spacing(2),
+          px: theme.spacing(4),
+        }}
+      >
+        <Typography
+          sx={{
+            color: theme.palette.text.tertiary,
+            fontSize: 12,
+            fontWeight: 500,
+          }}
+        >
+          {formatDate(new Date(label))}
+        </Typography>
+        {Object.keys(config)
+          .reverse()
+          .map((key) => {
+            const { color } = config[key];
+            const dotColor = theme.palette[color].main;
+
+            return (
+              <Stack
+                key={`${key}-tooltip`}
+                direction="row"
+                alignItems="center"
+                gap={theme.spacing(3)}
+                mt={theme.spacing(1)}
+                sx={{
+                  "& span": {
+                    color: theme.palette.text.tertiary,
+                    fontSize: 11,
+                    fontWeight: 500,
+                  },
+                }}
+              >
+                <Box
+                  width={theme.spacing(4)}
+                  height={theme.spacing(4)}
+                  backgroundColor={dotColor}
+                  sx={{ borderRadius: "50%" }}
+                />
+                <Typography
+                  component="span"
+                  textTransform="capitalize"
+                  sx={{ opacity: 0.8 }}
+                >
+                  {config[key].text}
+                </Typography>{" "}
+                <Typography component="span">
+                  {payload[0].payload[key]}
+                </Typography>
+              </Stack>
+            );
+          })}
+      </Box>
+    );
+  }
+  return null;
+};
+
+CustomToolTip.propTypes = {
+  active: PropTypes.bool,
+  payload: PropTypes.array,
+  label: PropTypes.string,
+  config: PropTypes.object,
+};
+
+/**
+ * Processes data to insert gaps with null values based on the interval.
+ * @param {Array} data
+ * @param {number} interval - The interval in milliseconds for gaps.
+ * @returns {Array} The formatted data with gaps.
+ */
+const processDataWithGaps = (data, interval) => {
+  if (data.length === 0) return [];
+  let formattedData = [];
+  let last = new Date(data[0].createdAt).getTime();
+
+  // Helper function to add a null entry
+  const addNullEntry = (timestamp) => {
+    formattedData.push({
+      accessibility: "N/A",
+      bestPractices: "N/A",
+      performance: "N/A",
+      seo: "N/A",
+      createdAt: timestamp,
+    });
```

Comment:
What was the reason for interpolating missing data points?  Was the appearance of the graph being distorted?

It seems like users might not realize that these data points are actually missing rather than there just being no data for them if we display N/A.  Maybe it would be better not to interpolate the points, or if it is critical for appearance then maybe some indication that the check doesn't actually exist?

Reply:
The reason for that is because the x-axis ticks did not appear to be continuous. Since the monitor will run checks while the server is active, once you shut it down and then turn it back on, it'll start displaying checks where it initially left off, so you'll have cases where the previous check might've happened let's say 8 hours ago, which didn't feel right to me. So I added null data points for a more accurate representation of time. I don't think removing the interpolation all together is ideal but what we can do is remove the `connectNulls` prop which will not connect the dots anymore, leaving a visible gap in the chart. This should make it more intuitive for users but will hurt the overall aesthetics😭 

🤔 I also think we should probably add this kind of interpolation to our other charts.

A paused monitor resumed after several days:
![image](https://github.com/user-attachments/assets/ab5ea6bb-d4bb-4ed9-967b-82c6fda801e2)

![image](https://github.com/user-attachments/assets/54a161b8-bf37-4d62-84f7-d31439b957d1)